wgsl/index.bs

```diff
@@ -3378,24 +3431,24 @@ access is an <dfn noexport>invalid memory reference</dfn>.
     * if the loaded value is a vector, the value (0, 0, 0, x), where x is:
         * 0, 1, or the maximum positive value for integer components
         * 0.0 or 1.0 for floating-point components
-[=statement/assignment|Stores=] to an invalid reference may do one of:
+: <dfn noexport>Invalid Store</dfn>
+:: [=statement/assignment|Stores=] to an invalid reference may do one of:
     * when the [=originating variable=] is a [=storage buffer=],
         store the value to any [=memory locations|memory location(s)=] of the
         WebGPU {{GPUBuffer}} bound to the [=originating variable=]
     * when the [=originating variable=] is not a [=storage buffer=],
         store the value to any [=memory locations|memory locations(s)=] in the
         originating variable
-    * not be executed
-It is a [=dynamic error=] if [[#atomic-rmw|read-modify-write atomics]] that
-operate on an invalid memory reference load and store from different [=memory
-locations|memory locations=] if they access memory.
+    * not be executed.
 
-Note: An invalid memory reference can occur even if the bytes accessed would
-still be within memory locations of the originating variable.
-Each access into a [=composite=] type must be made by a valid memory reference.
-See [[#composite-value-decomposition-expr]].
+A data race may occur if an invalid load or store is redirected to access different
+locations inside a variable in a shared address space.
+For example, the accesses of several concurrently executing invocations may be redirected
+to the first element in an array.  If at least access is a write, and they are not
+otherwise synchronized, then the result is a data race, and hence a dynamic error.
```

Comment:
Needs additional paragraph:
may cause bad uniformity result.  Eg. hangs in compute shaders. bad derivatives in fragment shaders.
These are not analyzed.